addons/website_slides/models/slide_slide.py

```diff
@@ -877,16 +911,39 @@ def _action_set_completed(self, target_partner):
             'vote': 0,
             'completed': True} for new_slide in new_slides])
 
-        return True
+    def action_mark_uncompleted(self):
+        if any(not slide.can_self_mark_uncompleted for slide in self):
+            raise UserError(_('You cannot mark a slide as uncompleted if you are not among its members.'))
 
-    def _action_set_quiz_done(self):
-        if any(not slide.channel_id.is_member for slide in self):
-            raise UserError(_('You cannot mark a slide quiz as completed if you are not among its members.'))
+        if any(not slide.user_has_completed for slide in self):
+            raise UserError(_('This slide is already set as uncompleted'))
+
+        # Remove the Karma point gained
+        self._action_set_quiz_done(completed=False)
+
+        self.env['slide.slide.partner'].sudo().search([
+            ('slide_id', 'in', self.ids),
+            ('partner_id', '=', self.env.user.partner_id.id),
+        ]).completed = False
+
+    def _action_set_quiz_done(self, completed=True):
+        """Add or remove karma point related to the quiz.
+
+        :param completed: True if the quiz will be marked as completed (karma will be increased)
```

Comment:
Would add:
"If set to False, we will remove the karma instead of increasing it, so that the user can take the quiz multiple times but not gain karma infinitely". 

To avoid any confusion.